File: Cutting Rectangles.cpp
class Solution{
public:
    vector<long long int> minimumSquares(long long int L, long long int B)

    {

        // code here

        vector<long long>ans;

        long long k=__gcd(L,B);

        long long n=(L*B)/(k*k);

        ans.push_back(n);

        ans.push_back(k);

        return ans;

    }
};

File: Cutting Rectangles.java
class Solution{

    static long commanfactor(long A, long B ){

        if(B == 0) return A;

        return commanfactor(B,A%B);

    }

    static List<Long> minimumSquares(long L, long B)

    {

        // code here

        long dim = commanfactor(L,B); //dimension

        long nos = (L*B) / (dim*dim); // No.of squre

        List<Long> ls = new ArrayList<>();

        ls.add(nos);

        ls.add(dim);

        return ls;

    }

}
